tests/wiseagents/test_coordinator.py

```diff
@@ -1,54 +1,88 @@
-import logging
 import os
 import threading
-import pytest
 
-from wiseagents import WiseAgent, WiseAgentMessage, WiseAgentRegistry
+from wiseagents import WiseAgentMessage, WiseAgentRegistry
 from wiseagents.llm import OpenaiAPIWiseAgentLLM
 from wiseagents.transports import StompWiseAgentTransport
-from wiseagents.wise_agent_impl import LLMOnlyWiseAgent, PassThroughClientAgent, SequentialCoordinatorWiseAgent
+from wiseagents.wise_agent_impl import CollaboratorWiseAgent, CoordinatorWiseAgent, PassThroughClientAgent
 
 cond = threading.Condition()
 
 
-def response_delivered(message: WiseAgentMessage):
+def final_response_delivered(message: WiseAgentMessage):
     with cond:
         response = message.message
-        assert "Farah" in response
-        assert "Agent1" in response
-        assert "Agent2" in response
+        assert response
         print(f"C Response delivered: {response}")
         cond.notify()
 
-@pytest.mark.needsllama
-def test_sequential_coordinator():
+
+def test_coordinator():
     """
-    Requires STOMP_USER and STOMP_PASSWORD.
+    Requires STOMP_USER, STOMP_PASSWORD, and a Groq API_KEY.
     """
-    llm1 = OpenaiAPIWiseAgentLLM(system_message="Your name is Agent1. Answer my greeting saying Hello and my name and tell me your name.",
-                                 model_name="llama3.1", remote_address="http://localhost:11434/v1")
-    agent1 = LLMOnlyWiseAgent(name="Agent1", description="This is a test agent", llm=llm1,
-                              trasport=StompWiseAgentTransport(host='localhost', port=61616, agent_name="Agent1"))
+    groq_api_key = os.getenv("GROQ_API_KEY")
+    llm1 = OpenaiAPIWiseAgentLLM(system_message="You will be coordinating a group of agents to solve a problem.",
+                                 model_name="llama-3.1-70b-versatile",
+                                 remote_address="https://api.groq.com/openai/v1",
+                                 api_key=groq_api_key)
+    agent1 = CoordinatorWiseAgent(name="Coordinator", description="This is a coordinator agent", llm=llm1,
+                                  transport=StompWiseAgentTransport(host='localhost', port=61616, agent_name="Coordinator"),
+                                  phases=["Information Gathering", "Verification of Information", "Analysis"],
+                                  max_iterations=2)
+
+    llm4 = OpenaiAPIWiseAgentLLM(system_message="You are a helpful assistant that will provide information about the given error message",
```

Comment:
Do we really need to have multiple OpenaiAPIWiseAgentLLM. Having all the messages managed by the agent, we may inject the same llm to each agent. When using process_chat_completion also system_message is managed by the agent, so it could be ok to use the same LLm which is fully stateless using process_chat_completion.
An imporvement in terms of clarity could be to have a stateless vs staefull OpenAIApi implementation. I've opened this (not assigned to anyone at the moment) for this: [#213](https://github.com/wise-agents/wise-agents/issues/213)

Reply:
We could do that, we'd just need to add the ability to specify an optional system_message to agents. I've created https://github.com/wise-agents/wise-agents/issues/217 to track this.